test/test_summarizer.py

```diff
@@ -94,3 +94,49 @@ def test_summarization_pipeline_one_summary(document_store, retriever, summarize
     answers = output["answers"]
     assert len(answers) == 1
     assert answers[0]["answer"] in EXPECTED_ONE_SUMMARIES
+
+@pytest.mark.slow
+@pytest.mark.summarizer
+def add_metadata_summerizer():
```

Comment:
The concept is fine, but I think this test should be split into two units.

For example, how about: 
```python
def add_metadata_summarizer_no_preexisting_context():
    document = Document( ... second document ... )

    summarizer = TransformersSummarizer(model_name_or_path="google/pegasus-xsum")
    summary = summarizer.predict(documents=[deepcopy(document)])
    
    document_dict = document.dict()
    summary_dict = summary.dict()

    assert list(document_dict.keys()) ==  list(summary_dict.keys()) - ["context"]
    for key, value in document_dict.items():
        assert document_dict[key] == summary_dict[key]


def add_metadata_summarizer_overwrite_context():
    document = Document( ... first document ... )

    summarizer = TransformersSummarizer(model_name_or_path="google/pegasus-xsum")
    summary = summarizer.predict(documents=[deepcopy(document)])
    
    document_dict = document.dict()
    summary_dict = summary.dict()

    assert list(document_dict.keys()) ==  list(summary_dict.keys())
    for key, value in document_dict.items():
        if key == "context":
            assert document_dict[key] != summary_dict[key]
       else:
            assert document_dict[key] == summary_dict[key]
```

In addition, we definitely need to test what happens if `generate_single_summary` is True.